bundles/org.openhab.binding.guntamatic/src/main/java/org/openhab/binding/guntamatic/internal/GuntamaticHandler.java

```diff
@@ -0,0 +1,426 @@
+                            pattern = "%.2f %unit%";
+                        }
+                    }
+
+                    ChannelTypeUID channelTypeUID = new ChannelTypeUID(BINDING_ID, channel);
+                    guntamaticChannelTypeProvider.addChannelType(channelTypeUID, channel, itemType,
+                            "Guntamatic " + label, false, pattern);
+                    Channel newChannel = ChannelBuilder.create(new ChannelUID(thing.getUID(), channel), itemType)
+                            .withType(channelTypeUID).withKind(ChannelKind.STATE).withLabel(label).build();
+                    channelList.add(newChannel);
+                    channels.put(i, channel);
+                    units.put(i, unit);
+
+                    logger.debug(
+                            "Supported Channel: Idx: '{}', Name: '{}'/'{}', Type: '{}'/'{}', Unit: '{}', Pattern '{}' ",
+                            String.format("%03d", i), label, channel, type, itemType, unit, pattern);
+
+                    /*
+                     * logger.debug("Supported Channel: Idx: {}, Name: {}, Type: {}", String.format("%03d", i), channel,
+                     * itemType);
+                     */
+                }
+            }
+        }
+        ThingBuilder thingBuilder = editThing();
+        thingBuilder.withoutChannels(channelList);
+        thingBuilder.withChannels(channelList);
+        updateThing(thingBuilder.build());
+        initalized = true;
+    }
+
+    private static String guessQuantityType(String type, String unit) {
+        String quantityType;
+
+        if (type.isEmpty()) {
+            type = "Number";
+        }
+
+        if ("%".equals(unit)) {
+            quantityType = type + ":Dimensionless";
+        } else if ("°C".equals(unit) || "°F".equals(unit)) {
+            quantityType = type + ":Temperature";
+        } else if ("m³".equals(unit)) {
+            quantityType = type + ":Volume";
+        } else if ("d".equals(unit) || "h".equals(unit)) {
+            quantityType = type + ":Time";
+        } else {
+            quantityType = type;
+        }
+
+        return quantityType;
+    }
+
+    private static String replaceUmlaut(String input) {
+
+        // replace all lower Umlauts
+        String output = input.replace("ü", "ue").replace("ö", "oe").replace("ä", "ae").replace("ß", "ss");
+
+        // first replace all capital umlaute in a non-capitalized context (e.g. Übung)
+        output = output.replaceAll("Ü(?=[a-zäöüß ])", "Ue").replaceAll("Ö(?=[a-zäöüß ])", "Oe")
+                .replaceAll("Ä(?=[a-zäöüß ])", "Ae");
+
+        // now replace all the other capital umlaute
+        output = output.replace("Ü", "UE").replace("Ö", "OE").replace("Ä", "AE");
+
+        return output;
+    }
+
+    private String toLowerCamelCase(String string) {
+        char delimiter = ' ';
+        string = string.replace("´", "").replaceAll("[^\\w]", String.valueOf(delimiter));
```

Comment:
By security, I would suggest to not update the input method parameter.

Reply:
```
    private String toLowerCamelCase(String input) {
        const char delimiter = ' ';
        String output = input.replace("´", "").replaceAll("[^\\w]", String.valueOf(delimiter));
```